contracts/modules/delegates/IexecPoco2Delegate.sol

```diff
@@ -25,574 +10,541 @@ import "./SignatureVerifier.sol";
+        IexecLibCore_v5.Contribution storage contribution = m_contributions[_taskid][_msgSender()];
+        IexecLibCore_v5.Deal memory deal = m_deals[task.dealid];
+
+        require(task.status == IexecLibCore_v5.TaskStatusEnum.ACTIVE);
+        require(task.contributionDeadline > now);
+        require(task.contributors.length == 0);
+        require(deal.trust == 1); // TODO / FUTURE FEATURE: consider sender's score ?
+
+        bytes32 resultHash = keccak256(abi.encodePacked(_taskid, _resultDigest));
+        bytes32 resultSeal = keccak256(abi.encodePacked(_msgSender(), _taskid, _resultDigest));
+
+        require(
+            (deal.callback == address(0) && _resultsCallback.length == 0) ||
+                keccak256(_resultsCallback) == task.resultDigest
+        );
+
+        // need enclave challenge if tag is set
+        require(_enclaveChallenge != address(0) || (deal.tag[31] & 0x01 == 0));
+
+        // Check that the worker + taskid + enclave combo is authorized to contribute (scheduler signature)
+        require(
+            _checkSignature(
+                (_enclaveChallenge != address(0) && m_teebroker != address(0))
+                    ? m_teebroker
+                    : deal.workerpool.owner,
+                _toEthSignedMessage(
+                    keccak256(abi.encodePacked(_msgSender(), _taskid, _enclaveChallenge))
+                ),
+                _authorizationSign
+            )
+        );
+
+        // Check enclave signature
+        require(
+            _enclaveChallenge == address(0) ||
+                _checkSignature(
+                    _enclaveChallenge,
+                    _toEthSignedMessage(keccak256(abi.encodePacked(resultHash, resultSeal))),
+                    _enclaveSign
+                )
+        );
+
+        contribution.status = IexecLibCore_v5.ContributionStatusEnum.PROVED;
+        contribution.resultHash = resultHash;
+        contribution.resultSeal = resultSeal;
+        contribution.enclaveChallenge = _enclaveChallenge;
+
+        task.status = IexecLibCore_v5.TaskStatusEnum.COMPLETED;
+        task.consensusValue = contribution.resultHash;
+        task.revealDeadline = task.timeref.mul(REVEAL_DEADLINE_RATIO).add(now);
+        task.revealCounter = 1;
+        task.winnerCounter = 1;
+        task.resultDigest = _resultDigest;
+        task.results = _results;
+        task.resultsCallback = _resultsCallback; // Expansion - result separation
+        task.contributors.push(_msgSender());
+
+        successWork(task.dealid, _taskid);
+        distributeRewardsFast(_taskid);
+
+        emit TaskContribute(_taskid, _msgSender(), resultHash);
+        emit TaskConsensus(_taskid, resultHash);
+        emit TaskReveal(_taskid, _msgSender(), _resultDigest);
+        emit TaskFinalize(_taskid, _results);
+
+        executeCallback(_taskid, _resultsCallback);
+    }
+
+    function reveal(
+        bytes32 _taskid,
+        bytes32 _resultDigest // worker
```

Comment:
Can you move comment?

Reply:
Deleted
[Fix comment](https://github.com/iExecBlockchainComputing/PoCo/pull/54/commits/f2ac8eab73bcbb26d95ba0387bf715fc579eb28d)